ncrypt/pgpkey.c

```diff
@@ -444,7 +444,7 @@ struct PgpKeyInfo *pgp_getkeybyaddr(struct Address *a, KeyFlags abilities,
 
   if (matches)
   {
-    if (oppenc_mode)
+    if (oppenc_mode || !isatty(STDIN_FILENO))
```

Comment:
These are a bit ugly, especially so deep in the code.
I think we can set `oppenc_mode` instead.
(I'll post a backtrace in a sec)

Reply:
I tried something similar initially: adding a `bool batch_mode` parameter that would be passed all across the stack.  At some point (function pointers), I found out that I couldn't change some function prototypes.  Then I thought of modifying oppenc_mode, but it is also passed through so many layers that it was scary, and sometimes it's hardcoded to 0 or 1, so I was worried that it might have consequences somewhere else.  `isatty(STDIN_FILENO)` was the smallest change I could do.

I'll give another try at modifying oppenc_mode and will check the consequences.